### PR TITLE
Feat: Implement `create_table_if_not_exists`

### PR DESCRIPTION
**Pull request by hussein-awala**
closes: #406

This PR implements the `create_table_if_not_exists` method to create the table if it doesn't exist, and return the existing table if it exists instead of raising a `TableAlreadyExistsError` exception.

### CODE REVIEW

**Comment by Fokko** [comment]
I think we should have some more discussion around this. I would prefer to have, analog to SQL, a `CREATE OR REPLACE` API.

**Comment by syun64** [comment]
> I think we should have some more discussion around this. I would prefer to have, analog to SQL, a `CREATE OR REPLACE` API.

This feels analogous to the SQL `CREATE TABLE IF NOT EXISTS`, which allows applications to run the statement idempotently in a Production deployed code without throwing an exception. Hence I'd argue it fills in a different niche than CREATE OR REPLACE, which effectively replaces an existing table (which is exactly what we want to avoid doing).

**Comment by hussein-awala** [comment]
Exactly, this implements `CREATE TABLE IF NOT EXISTS` and not `CREATE OR REPLACE`.

If you prefer to have a new method `create_table_if_not_exists`, I can move the table creation block to a new private/protected method, and use it in `create_table` and the new `create_table_if_not_exists`.

**Comment by hussein-awala** [comment]
I will update the PR then 👍

**Comment by hussein-awala** [comment]
I think it's ready 🤞

File: tests/catalog/test_rest.py [hunk 1]
```diff
@@ -560,6 +560,64 @@ def test_create_table_409(rest_mock: Mocker, table_schema_simple: Schema) -> Non
     assert "Table already exists" in str(e.value)
 
 
+def test_create_table_if_not_exists_200(
```
Comment (Fokko) [inline tests/catalog/test_rest.py:563]: Nice! It would have been easier to write an integration test, but great job on the mocking 👍

**Review by Fokko — APPROVED** [review]
@hussein-awala This looks great, thanks!

**Pull request merged**